ts/features/fci/components/DocumentWithSignature.tsx

```diff
@@ -0,0 +1,315 @@
+      // TODO: refactor this function to use fp-ts
+      const doc = documents[currentDoc];
+      const url = doc.url;
+
+      const existingPdfBytes = await ReactNativeBlobUtil.fetch("GET", url).then(
+        res => res.base64()
+      );
+
+      await PDFDocument.load(pdfFromBase64(existingPdfBytes)).then(res => {
+        const page = attrs.page;
+        // eslint-disable-next-line @typescript-eslint/restrict-plus-operands
+        setSignaturePage(page.valueOf() + 1);
+        // The signature box is drawn using the coordinates of the signature field.
+        res.getPage(page).drawRectangle({
+          x: attrs.bottom_left.x ?? 0,
+          y: attrs.bottom_left.y ?? 0,
+          height: Math.abs(
+            (attrs.top_right.y ?? 0) - (attrs.bottom_left.y ?? 0)
+          ),
+          width: Math.abs(
+            (attrs.top_right.x ?? 0) - (attrs.bottom_left.x ?? 0)
+          ),
+          color: rgb(0, 0.77, 0.79),
+          opacity: 0.5,
+          borderOpacity: 0.75
+        });
+
+        return res.saveAsBase64().then(r => setPdfString(pdfFromBase64(r)));
+      });
+    },
+    [documents, currentDoc]
+  );
+
+  const onSignatureDetail = React.useCallback(
+    (attrs: ExistingSignatureFieldAttrs | SignatureFieldToBeCreatedAttrs) => {
+      if (hasUniqueName(attrs)) {
+        drawRectangleOverSignatureFieldById(attrs.unique_name).catch(
+          readableReport // TODO: it should be displayed to the user?
+        );
+      } else {
+        drawRectangleOverSignatureFieldByCoordinates(attrs).catch(
+          readableReport // TODO: it should be displayed to the user?
+        );
+      }
+    },
+    [
+      drawRectangleOverSignatureFieldByCoordinates,
+      drawRectangleOverSignatureFieldById
+    ]
+  );
+
+  React.useEffect(() => {
+    pipe(
+      attrs,
+      O.fromNullable,
+      O.map(_ => {
+        onSignatureDetail(_);
+      }),
+      O.getOrElse(() => {
+        setPdfString("");
+        setSignaturePage(0);
+      })
+    );
+  }, [attrs, onSignatureDetail]);
+
+  React.useEffect(() => {
+    if (isPdfLoaded) {
+      pipe(
+        pdfRef.current,
+        O.fromNullable,
+        O.map(_ => _.setPage(signaturePage))
+      );
+    }
+  }, [pdfRef, signaturePage, isPdfLoaded]);
+
+  const onContinuePress = () => props.onClose();
+
+  const continueButtonProps = {
+    block: true,
+    primary: true,
+    onPress: onContinuePress,
+    title: I18n.t("features.fci.documents.footer.backToSignFieldsList")
+  };
+
+  const pointToPage = (page: number) =>
+    pipe(
+      pdfRef.current,
+      O.fromNullable,
+      O.map(_ => _.setPage(page))
+    );
+
+  const renderPager = () => (
+    <Pdf
+      ref={pdfRef}
+      source={{
+        uri: pdfString
+      }}
+      onLoadComplete={(numberOfPages, _) => {
+        setTotalPages(numberOfPages);
+        setIsPdfLoaded(true);
+      }}
+      onPageChanged={(page, _) => {
+        setCurrentPage(page);
+      }}
+      onError={constNull}
```

Comment:
How do we handle errors on this components?

Reply:
Added a prop.onError because modal so we catch the error on parent screen.